torch/onnx/symbolic_opset9.py

```diff
@@ -529,6 +529,25 @@ def reciprocal(g: jit_utils.GraphContext, self):
 @_beartype.beartype
 def cat(g: jit_utils.GraphContext, tensor_list, dim):
     tensors = symbolic_helper._unpack_list(tensor_list)
+    # torch.cat ignores empty tensors such as `torch.Tensor([])`
+    # These needs to be removed as input from ONNX's concat too, otherwise shape inference
+    # will likely fail due to inputs with different ranks (0 for empty tensor, > 0 for anything else)
+    nonempty_tensors = []
+    for t in tensors:
+        if symbolic_helper._is_constant(t) and not symbolic_helper._get_tensor_dim_size(t, 0):
+            continue
+        nonempty_tensors.append(t)
+    assert len(nonempty_tensors) > 0
+    assert all(
+        [
+            symbolic_helper._get_tensor_rank(t)
```

Comment:
`_get_tensor_rank` returns None when rank is unknown. Is it expected to raise assertion error for this case?

Reply:
This assert only makes sense if `symbolic_helper._get_tensor_rank(nonempty_tensors[0] is not None`, so I will change it accordingly